models/logs.js

```diff
@@ -35,10 +35,22 @@ const fetchLogs = async (query, param) => {
     let call = logsModel.where("type", "==", param);
     Object.keys(query).forEach((key) => {
       // eslint-disable-next-line security/detect-object-injection
-      call = call.where(key, "==", query[key]);
+      if (key !== "limit" && key !== "lastDocId") {
+        call = call.where(key, "==", query[key]);
+      }
     });
 
-    const snapshot = await call.get();
+    const { limit, lastDocId } = query;
+    let lastDoc = "";
```

Comment:
why do we need to do that? does it give any error if it is undefined?

Reply:
yes, we cannot use undefined as a firestore value. Refactored the code using nullish coalescing